needy/cache.py

```diff
@@ -0,0 +1,29 @@
+
+class Cache:
+    def __init__(self, configuration={}):
+        self.__configuration = configuration
+
+    @staticmethod
+    def type():
+        '''type of cache'''
+        raise NotImplementedError('type')
+
+    def description(self):
+        '''uri, path, or other configuration description'''
+        raise NotImplementedError('description')
+
+    def configuration(self):
+        '''configuration dictionary to serialize to configuration files'''
+        return self.__configuration
+
+    def archive(self, source, key):
+        '''make files and folders at source retrievable with the given key'''
+        raise NotImplementedError('archive')
+
+    def unarchive(self, destination, key):
```

Comment:
Alternatively, you could use `load_directory` and `store_directory` methods, which would bundle things up and cache them via `load_file` / `store_file` methods. I like the load / store naming better, and this would require less of the implementations since they'd only need methods for working with individual files.